save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/UsersController.kt

```diff
@@ -44,11 +47,12 @@ class UsersController(
     @PostMapping("/new")
     fun saveNewUser(@RequestBody user: User) {
         val userName = requireNotNull(user.name) { "Provided user $user doesn't have a name" }
-        userRepository.findByName(userName)?.let {
+        originalLoginRepository.findByName(userName)?.user?.let {
```

Comment:
Why are you checking `originalLoginRepository` here? What if we need to save a user with a name that exists in original logins but from a new authorization source?